tests/test_cffi.py

```diff
@@ -15,81 +15,81 @@ def test_cffi(self):
             self.assertEqual(lib.monetdbe_type_unknown, 14)
 
     def test_append_too_many_columns(self):
-        con = connect()
-        con.execute("CREATE TABLE test (i int)")
-        data = {'i': np.array([1, 2, 3]), 'j': np.array([1, 2, 3])}
-        with self.assertRaises(ProgrammingError):
-            con._internal.append(table='test', data=data)
+        with connect() as con:
+            con.execute("CREATE TABLE test (i int)")
+            data = {'i': np.array([1, 2, 3]), 'j': np.array([1, 2, 3])}
+            with self.assertRaises(ProgrammingError):
+                con._internal.append(table='test', data=data)
```

Comment:
this all comes from the sqlite test suite, which assumes the connections are closed by the garbage collector at some point.

but yes it is much better be explicit and use a context manager